crates/rpc/src/routes/beacon.rs

```diff
@@ -28,7 +34,23 @@ pub fn get_beacon_routes(
         .and_then(move || get_genesis(network_spec.genesis.clone()))
         .with(log("genesis"));
 
+    let randao = {
+        let db = db.clone();
+        beacon_base
+            .and(path("states"))
+            .and(param::<ID>())
+            .and(path("randao"))
+            .and(query::<RandaoQuery>())
+            .and(end())
+            .and(get())
+            .and_then(move |state_id: ID, query: RandaoQuery| {
+                let epoch = query.epoch;
+                get_randao_mix(state_id, epoch, db.clone())
```

Comment:
```suggestion
        let db = db.clone();
        beacon_base
            .and(path("states"))
            .and(param::<ID>())
            .and(path("randao"))
            .and(query::<RandaoQuery>())
            .and(end())
            .and(get())
            .and_then(move |state_id: ID, query: RandaoQuery| {
                get_randao_mix(state_id, query, db)
```
we don't need the clone here because it is cloned above

also just pass `query` in

Reply:
So i did try removing the clone inside but it doesnt work, it says i need to clone both inside and outisde. i think its an issue with moving into the closure the FnOnce vs Fn

making a workaround using filters